packages/react/src/primitives/Checkbox/useCheckbox.ts

```diff
@@ -4,8 +4,8 @@ import { isFunction } from '../shared/utils';
 import { UseCheckbox } from '../types/checkbox';
 
 export const useCheckbox = (
-  initialChecked: boolean,
-  onChangeProp: React.ChangeEventHandler
+  initialChecked: boolean | undefined,
```

Comment:
Can we just keep this as `boolean` and assign a default value of `false`? _Think_ it would allow `setDataChecked` to have a type of `Dispatch<SetStateAction<boolean>>` in the `UseCheckbox` interface

Reply:
I'd rather now change the existing behavior, in particular when it comes to the uncontrolled Checkbox. I checked current behavior and checked is initially `undefined` when it hasn't been interacted with.